### PR TITLE
rename CSI_REMOVE_HOLDER_PODS to CSI_DISABLE_HOLDER_PODS

### PR DESCRIPTION
**Pull request by BlaineEXE**
I made a mistake when reporting the requirements for ocs-operator to Nitin. In Rook, the true config value is named
CSI_DISABLE_HOLDER_PODS. Ensure all references are updated to reflect this.

If this is not change, the Multus feature to disable holder pods won't work properly. Apologies for the confusion.

### CODE REVIEW

**Review by Nikhil-Ladha — COMMENTED** [review]
But, we still need this key in the cm as that's part of the rook CSV and, if I am not wrong being used by provider mode as well.

**Review by iamniting — CHANGES_REQUESTED** [review]
@subhamkrai @Nikhil-Ladha  Can we change the ref of this key in the rook CSV and get the image.

Pls merge the PR only in the main for now in the rook as well. We should merge both PRs together in the release-4.16 branch so that we do not break the DS pipeline.

@BlaineEXE once we have the rook bundle image pls update the bundle image as well within this PR.

File: controllers/util/k8sutil.go [hunk 1]
```diff
@@ -37,7 +37,7 @@ const (
 	EnableTopologyKey           = "CSI_ENABLE_TOPOLOGY"
 	TopologyDomainLabelsKey     = "CSI_TOPOLOGY_DOMAIN_LABELS"
 	EnableNFSKey                = "ROOK_CSI_ENABLE_NFS"
-	CsiRemoveHolderPodsKey      = "CSI_REMOVE_HOLDER_PODS"
+	CsiRemoveHolderPodsKey      = "CSI_DISABLE_HOLDER_PODS"
```
Comment (Nikhil-Ladha) [inline controllers/util/k8sutil.go:40]: I think we should also change the variable name to `CsiDisableHolderPodsKey`.

**Comment by Nikhil-Ladha** [comment]
> CSI_DISABLE_HOLDER_PODS

Here's the PR for the rook repo: https://github.com/red-hat-storage/rook/pull/618
Once, the PR is merged we should be having a new bundle image at `quay.io/ocs-dev/rook-ceph-operator-bundle` repo.

File: controllers/util/k8sutil.go (hunk 1 above)
Comment (BlaineEXE) [reply controllers/util/k8sutil.go]: Updated.

**Comment by BlaineEXE** [comment]
> > CSI_DISABLE_HOLDER_PODS
> 
> Here's the PR for the rook repo: [red-hat-storage/rook#618](https://github.com/red-hat-storage/rook/pull/618) Once, the PR is merged we should be having a new bundle image at `quay.io/ocs-dev/rook-ceph-operator-bundle` repo.

Merged, and backport is waiting here: https://github.com/red-hat-storage/rook/pull/620

I haven't merged the backport b/c of this comment:

> Pls merge the PR only in the main for now in the rook as well. We should merge both PRs together in the release-4.16 branch so that we do not break the DS pipeline.
>
> [Blaine], once we have the rook bundle image pls update the bundle image as well within this PR.

Do I need to make changes to this PR to get the latest bundle? I'm not sure the exact workflow.

**Comment by iamniting** [comment]
> Do I need to make changes to this PR to get the latest bundle? I'm not sure the exact workflow.

@BlaineEXE yes we need to update the bundle image within this PR itself so that CI passes.

**Review by iamniting — CHANGES_REQUESTED** [review]
Pls update the bundle image here https://github.com/red-hat-storage/ocs-operator/blob/main/hack/common.sh#L94

**Comment by iamniting** [comment]
@Nikhil-Ladha Can you pls help @BlaineEXE with the image?

**Comment by BlaineEXE** [comment]
https://bugzilla.redhat.com/show_bug.cgi?id=2274548

**Comment by Nikhil-Ladha** [comment]
> 9e5499216

Yeah, that bundle image was pushed manually and it has the last few chars of the commit hash instead of the first few. That's the reason you are not able to grep it.
Try `git log --pretty=full pretty=oneline | grep "9e5499216"`, you will be able to find the commit.

But, after the recent update of the build workflow under `red-hat-storage/rook` to auto push the bundle images after each PR is merged, we are now using the first few chars of the commit hash.
So, if you check here, https://quay.io/repository/ocs-dev/rook-ceph-operator-bundle?tab=tags, it has the bundle image `master-f8e5f814c` that was pushed after my PR was merged.

**Comment by BlaineEXE** [comment]
Updated with new Rook image now

**Review by iamniting — APPROVED** [review]

**Pull request merged**